src/domain.cc

```diff
@@ -1353,6 +1353,160 @@ NLV_WORKER_EXECUTE(Domain, SendKeys)
   data_ = true;
 }
 
+NAN_METHOD(Domain::Migrate)
+{
+  HandleScope scope;
+  unsigned long flags = 0;
+  unsigned long bandwidth = 0;
+
+  if(args.Length() < 2 ||
+      (!args[0]->IsObject() && !args[1]->IsFunction())) {
+    NanThrowTypeError("you must specify an object and a callback");
+    NanReturnUndefined();
+  }
+
+  Local<Object> args_ = args[0]->ToObject();
+
+  if(!args_->Has(NanNew("dest_uri"))) {
+    NanThrowTypeError("You must have set property dest_uri in the object");
+    NanReturnUndefined();
+  }
+
+  std::string dest_uri(*NanUtf8String(args_->Get(NanNew("dest_uri"))));
+  std::string dest_name(*NanUtf8String(args_->Get(NanNew("dest_name"))));
+
+  if(args_->Has(NanNew("flags"))) {
+    Local<Array> flags_ = args_->Get(NanNew("flags")).As<Array>();
+    unsigned int length = flags_->Length();
+    for (unsigned int i = 0; i < length; i++)
+      flags |= flags_->Get(Integer::New(i))->Int32Value();
+  }
+
+  if(args_->Has(NanNew("bandwidth"))) {
+    bandwidth = args_->Get(NanNew("bandwidth"))->Int32Value();
+  }
+
+  Domain *domain = ObjectWrap::Unwrap<Domain>(args.This());
+
+  NanCallback *callback = new NanCallback(args[1].As<Function>());
+  MigrateWorker *worker;
+
+  if(args_->Has(NanNew("dest_hypervisor"))) {
+    Local<Object> hyp_obj = args_->Get(NanNew("dest_hypervisor"))->ToObject();
+    if(!NanHasInstance(Hypervisor::constructor_template, hyp_obj)) {
+      NanThrowTypeError("You must specify a Hypervisor object instance");
+      NanReturnUndefined();
+    }
+
+    Hypervisor *hypervisor = ObjectWrap::Unwrap<Hypervisor>(hyp_obj);
+    worker = new MigrateWorker(callback, domain->handle_, hypervisor->handle_);
+  } else {
+    worker = new MigrateWorker(callback, domain->handle_, dest_uri);
+  }
+
+  worker->setBandwidth(bandwidth);
+  worker->setFlags(flags);
+  worker->setDestname(dest_name);
+  NanAsyncQueueWorker(worker);
+  NanReturnUndefined();
+}
+
+NLV_WORKER_EXECUTE(Domain, Migrate)
+{
+  NLV_WORKER_ASSERT_DOMAIN();
+
+  if(conn_) {
+    migrated_ = virDomainMigrate(Handle().ToDomain(), conn_, flags_, destname_.c_str(), uri_.c_str(), bandwidth_);
+    if(migrated_ == NULL) {
+      NLV_SET_LV_ERROR();
```

Comment:
I dig this refactoring, but can you explicitly put ```return;``` below it, so that we know that we're bailing? Even in cases like this where obviously nothing else is happening, I think it really helps readability